### PR TITLE
Feature/check task

### PR DESCRIPTION
**Pull request by theo-damiani**
Hey!
So in this PR, I added the logic to manage the boolean Done of a task.
I also refactor the way we delete/modify tasks because, in a previous way, there are a lot of problems when you scroll for example if you have a lot of tasks.
And also now when you click on a task, it opens a small panel to update it. And so it will be easier to add features for the future. For example, I wanted to add a boolean "prioritize" so when a task has this flag it is displayed on the list selection view.

I also try to add design according to what Yago did.

### CODE REVIEW

File: app/src/main/java/com/github/steroidteam/todolist/model/todo/Task.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 public class Task {
 
     private String body;
+    private Boolean done;
```
Comment (shoutizix) [inline app/src/main/java/com/github/steroidteam/todolist/model/todo/Task.java:14]: Maybe isDone would be a better name for the boolean

File: app/src/main/java/com/github/steroidteam/todolist/model/todo/Task.java [hunk 2]
```diff
@@ -55,4 +57,12 @@ public boolean equals(Object o) {
         Task task = (Task) o;
         return Objects.equals(body, task.body);
     }
+
+    public Boolean isDone() {
+        return done;
+    }
+
+    public void setDone(Boolean done) {
```
Comment (shoutizix) [inline app/src/main/java/com/github/steroidteam/todolist/model/todo/Task.java:65]: Same here, maybe "setIsDone" and "getIsDone" would be better for the boolean but maybe it's only a matter of personal taste haha

**Review by shoutizix — APPROVED** [review]
Hey,

It looks very good, I've tried it on my phone and the new panel to update the tasks looks great to me too !

As I mentionned the only little thing that maybe you could improve was the name of the boolean so that it starts with a verb but other than that it seems really great nice job !

File: app/src/main/java/com/github/steroidteam/todolist/view/adapter/TodoAdapter.java [hunk 3]
```diff
@@ -56,55 +67,56 @@ public void setTodoList(TodoList todoList) {
...
-    }
-
-    public void setCurrentlyDisplayedUpdateLayoutPos(Integer currentlyDisplayedUpdateLayoutPos) {
-        this.currentlyDisplayedUpdateLayoutPos = currentlyDisplayedUpdateLayoutPos;
-    }
-
     public interface TaskCustomListener {
-        void onClickCustom(int position);
+        void onClickCustom(TaskHolder holder, int position);
+        // void onFocusChangedCustom(int position, boolean hasFocus);
```
Comment (VPellenc) [inline app/src/main/java/com/github/steroidteam/todolist/view/adapter/TodoAdapter.java:72]: Commented-out code (not sure if intentional)

**Review by VPellenc — APPROVED** [review]
Tested it and it looks very nice!

One thing I noticed, when the panel is opened after clicking a task, it is not small on my emulator, it takes all the free space.
There was also the idea of having the Todo go to the bottom when getting marked as done but I think it looks great as is.

**Comment by theo-damiani** [comment]
> 
> 
> Tested it and it looks very nice!
> 
> One thing I noticed, when the panel is opened after clicking a task, it is not small on my emulator, it takes all the free space.
> There was also the idea of having the Todo go to the bottom when getting marked as done but I think it looks great as is.

Thanks, @VPellenc for the review!
Normally the panel looks like this: 


Could you confirm for me that you have the same thing?
This is not a dialog so yes it was built to take a big part of the screen (the same as the recycler view).

Yes, I didn't implement the logic to move a task to the bottom when it is done, because I will prefer to implement this in a PR that takes care of all the sorting aspect of a task as it is quite the same thing.

**Comment by VPellenc** [comment]
> > Tested it and it looks very nice!
> > One thing I noticed, when the panel is opened after clicking a task, it is not small on my emulator, it takes all the free space.
> > There was also the idea of having the Todo go to the bottom when getting marked as done but I think it looks great as is.
> 
> Thanks, @VPellenc for the review!
> Normally the panel looks like this:
> 
> 
> Could you confirm for me that you have the same thing?
> This is not a dialog so yes it was built to take a big part of the screen (the same as the recycler view).
> 
> Yes, I didn't implement the logic to move a task to the bottom when it is done, because I will prefer to implement this in a PR that takes care of all the sorting aspect of a task as it is quite the same thing.

Yes I have the same thing.
It looks a bit weird with the empty space until the Delete button but it's definitely not a blocker.
I could imagine it as an "extension" of an item in the recyclerView, where you can still see the other tasks around but I don't know how hard it would be to implement (if even possible). Either way I think it looks good enough for now.

**Comment by theo-damiani** [comment]
> Yes I have the same thing.
> It looks a bit weird with the empty space until the Delete button but it's definitely not a blocker.
> I could imagine it as an "extension" of an item in the recyclerView, where you can still see the other tasks around but I don't know how hard it would be to implement (if even possible). Either way I think it looks good enough for now.

I designed that way because I thought that we will have space to add more features: like add categories to a task, add a boolean to know if a task is important, ...

**Comment by theo-damiani** [comment]
Is it okay if we merge this PR?

**Comment by VPellenc** [comment]
> > Yes I have the same thing.
> > It looks a bit weird with the empty space until the Delete button but it's definitely not a blocker.
> > I could imagine it as an "extension" of an item in the recyclerView, where you can still see the other tasks around but I don't know how hard it would be to implement (if even possible). Either way I think it looks good enough for now.
> 
> I designed that way because I thought that we will have space to add more features: like add categories to a task, add a boolean to know if a task is important, ...

Yeah I see in that case my alternative makes less sense



> Is it okay if we merge this PR?

Good for me

**Pull request merged**